### PR TITLE
feat(stream): one small step for watermark

### PR DESCRIPTION
**Pull request by st1page**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
introduce the watermark in the proto and stream message
## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#6043

### CODE REVIEW

**Comment by yuhao-su** [comment]
LGTM!

File: src/stream/src/executor/mod.rs [hunk 1]
```diff
@@ -555,6 +585,7 @@ impl Message {
                 StreamMessage::StreamChunk(prost_stream_chunk)
             }
             Self::Barrier(barrier) => StreamMessage::Barrier(barrier.clone().to_protobuf()),
+            Self::Watermark(_) => todo!(),
```
Comment (yuhao-su) [inline src/stream/src/executor/mod.rs:588]: miss a issue link. But not a big deal.

**Review by yuhao-su — APPROVED** [review]

File: src/stream/src/executor/hash_join.rs [hunk 2]
```diff
@@ -659,6 +659,11 @@ impl<K: HashKey, S: StateStore, const T: JoinTypePrimitive> HashJoinExecutor<K,
                         self.chunk_size,
                     ) {
                         yield chunk.map(|v| match v {
+                            Message::Watermark(_) => {
+                                unimplemented!(
```
Comment (BugenZhao) [inline src/stream/src/executor/hash_join.rs:663]:
```suggestion
                                todo!(
```

Same for other occurrences.

File: proto/stream_plan.proto [hunk 3]
```diff
@@ -94,6 +94,13 @@ message Barrier {
   repeated uint32 passed_actors = 255;
 }
 
+message Watermark {
+  // the watermark column's index in the schema
```
Comment (BugenZhao) [inline proto/stream_plan.proto:98]: How about specifying "which schema", input, internal state, or output?

**Review by BugenZhao — COMMENTED** [review]
LGTM

**Review by BugenZhao — APPROVED** [review]
LGTM

**Pull request merged**